SocialCareCaseViewerApi.Tests/V1/UseCase/CaseStatus/CaseStatusExecutePostUseCaseTests.cs

```diff
@@ -0,0 +1,89 @@
+        private CaseStatusesUseCase _caseStatusesUseCase;
+        private CreateCaseStatusRequest _request;
+        private readonly CaseStatusType _typeInRequest = TestHelpers.CreateCaseStatusType();
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockDatabaseGateway = new Mock<IDatabaseGateway>();
+            _caseStatusesUseCase = new CaseStatusesUseCase(_mockDatabaseGateway.Object);
+
+            _mockDatabaseGateway.Setup(x => x.GetCaseStatusTypeWithFields(_typeInRequest.Name))
+                .Returns(_typeInRequest);
+
+            _request = CaseStatusHelper.CreateCaseStatusRequest(type: _typeInRequest.Name);
+
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(It.IsAny<int>()))
+                .Returns(TestHelpers.CreatePerson((int) _request.PersonId));
+
+            CaseStatus _caseStatus = CaseStatusHelper.CreateCaseStatus();
+
+            _mockDatabaseGateway.Setup(x => x.CreateCaseStatus(It.IsAny<CreateCaseStatusRequest>()))
+                .Returns(_caseStatus);
+
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(It.IsAny<string>()))
+                .Returns(TestHelpers.CreateWorker(email: _request.CreatedBy));
+        }
+
+        [Test]
+        public void CallsDatabaseGatewayToCheckPersonExists()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(It.IsAny<long>())).Returns(TestHelpers.CreatePerson((int) _request.PersonId));
+
+            _caseStatusesUseCase.ExecutePost(_request);
+            _mockDatabaseGateway.Verify(gateway => gateway.GetPersonByMosaicId(_request.PersonId));
+        }
+
+        [Test]
+        public void WhenPersonDoesNotExistThrowsPersonNotFoundException()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(It.IsAny<long>()));
+
+            Action act = () => _caseStatusesUseCase.ExecutePost(_request);
+
+            act.Should().Throw<PersonNotFoundException>()
+                .WithMessage($"'personId' with '{_request.PersonId}' was not found.");
+        }
+        [Test]
+        public void WhenTypeDoesNotExistThrowsTypeNotFoundException()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(It.IsAny<long>())).Returns(TestHelpers.CreatePerson((int) _request.PersonId));
+            _mockDatabaseGateway.Setup(x => x.GetCaseStatusTypeWithFields(It.IsAny<string>()));
+
+            Action act = () => _caseStatusesUseCase.ExecutePost(_request);
+
+            act.Should().Throw<CaseStatusTypeNotFoundException>();
+        }
+
+        [Test]
+        public void WhenCreatedByEmailDoesNotExistThrowsWorkerNotFoundException()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(It.IsAny<long>())).Returns(TestHelpers.CreatePerson((int) _request.PersonId));
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(It.IsAny<string>()))
+                .Returns((Worker) null);
```

Comment:
If you want a mock to return null you can just not have a .Returns() call after the setup, No need to cast null as a Worker